parser/transcript/src/main/java/de/danoeh/antennapod/parser/transcript/VttTranscriptParser.java

```diff
@@ -0,0 +1,145 @@
+package de.danoeh.antennapod.parser.transcript;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jsoup.Jsoup;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import de.danoeh.antennapod.model.feed.Transcript;
+import de.danoeh.antennapod.model.feed.TranscriptSegment;
+
+public class VttTranscriptParser {
+    private static final Pattern TIMESTAMP_PATTERN =
+            Pattern.compile("^(?:([0-9]{2}):)?([0-9]{2}):([0-9]{2})\\.([0-9]{3})$");
+
+    private static final Pattern VOICE_SPAN =
+            Pattern.compile("<v(?:\\.[^\\t\\n\\r &<>.]+)*[ \\t]([^\\n\\r&>]+)>");
+
+    private record Timings(long start, long end) {}
+
+    public static Transcript parse(String str) {
+        // This is basically a very light WebVTT parser.
+        // It uses WebVTT properties to be both exact and very light.
+        // We will only be parsing the WebVTT cue blocks.
+
+        if (StringUtils.isBlank(str)) {
+            return null;
+        }
+
+        // WebVTT line terminator can be \r\n, \n or \n, let's use only one
+        str = str.replaceAll("\r\n?", "\n");
+        List<String> lines = Arrays.asList(str.split("\n"));
+
+        Transcript transcript = new Transcript();
+        Iterator<String> iterator = lines.iterator();
+        Set<String> speakers = new HashSet<>();
+        String speaker = "";
+        TranscriptSegment segment = null;
+
+        // Iterate through cue blocks
+        while (iterator.hasNext()) {
+            String line = iterator.next();
+
+            if (!line.contains("-->")) {
+                continue;
+            }
+
+            Timings timings = parseCueTimings(line);
+            if (timings == null) {
+                return null; // Input is broken
+            }
+
+            String payload = parseCuePayload(iterator);
+
+            Matcher matcher = VOICE_SPAN.matcher(payload);
+            if (matcher.find()) {
+                speaker = matcher.group(1);
+                speakers.add(speaker);
+            }
+
+            payload = Jsoup.parse(payload).text(); // remove all HTML tags
+
+            // should we merge this segment with the previous one?
+            if (segment != null && segment.getSpeaker().equals(speaker)
+                    && timings.end - segment.getStartTime() < TranscriptParser.MAX_SPAN) {
+                segment = new TranscriptSegment(
+                        segment.getStartTime(), timings.end, segment.getWords() + " " + payload, speaker);
```

Comment:
This creates and then immediately drops quite a lot of segments. How about adding an `append(endTime, words)` method to the transcript segment that can add additional text? That could then maybe also be used to make Srt parsing a bit easier to read (instead of having a bunch of variables)

When first reading this, I was a bit confused why it creates a new segment object. An "append" method would be a bit easier to read for future developers as well

Reply:
It seems a good idea indeed.
I did not want to modify an external class at first, but indeed it would make things easier to read.
I won’t be able to access my dev setup next week but I’ll be able to this after.

Reply:
I added this method as requested